#!/usr/bin/env python

PACKAGE = "rbot280"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add( name,  paramtype, level, description, default, min, max)
gen.add("velKp", double_t, 0, 'linear vel P gain', 0.0, 0, 100)
gen.add("velKi", double_t, 0, 'linear vel I gain', 0.0, 0, 100)
gen.add("velKd", double_t, 0, 'linear vel D gain', 0.0, 0, 100)
gen.add("yawKp", double_t, 0, 'angular torque P gain', 0.6, 0, 100)
gen.add("yawKi", double_t, 0, 'angular torque I gain', 0.0, 0, 100)
gen.add("yawKd", double_t, 0, 'angular torque D gain', 0.0, 0, 100)
gen.add("maxSpeed", double_t, 0, 'max speed', 0.4, 0.0, 5.0)


exit(gen.generate(PACKAGE, "rbot280", "CmdVelPIDDynamic"))
